keylime/da/record.py

```diff
@@ -0,0 +1,375 @@
+
+        self.ps_url = config.get(self.svc, "persistent_store_url", fallback="")
+        self.tl_url = config.get(self.svc, "transparency_log_url", fallback="")
+        self.tsa_url = config.get(self.svc, "time_stamp_authority_url", fallback="")
+        self.tsa_cert = config.get(self.svc, "time_stamp_authority_certs_path", fallback="")
+
+        self.st_imp_path = ".".join(config.get(self.svc, "durable_attestation_import", fallback="").split(".")[0:-1])
+
+        self.rmv_a = ["ssl_context", "pending_event", "b64_encrypted_V"]
+        self.rcd_fmt = config.get(self.svc, "persistent_store_format", fallback="json")
+        self.rcd_enc = config.get(self.svc, "persistent_store_encoding", fallback="")
+        self.rcd_sa = config.get(self.svc, "transparency_log_sign_algo", fallback="sha256")
+        self.tmp_d_cl = True
+        self.key_tls_priv = None
+        self.key_tls_pub = key_tls_pub
+        self.priv_key = None
+
+    def record_create(self, agent_data, attestation_data, ima_policy_data, service="auto", signed_attributes="auto"):
+        _ = (agent_data, attestation_data, ima_policy_data, service, signed_attributes)
+
+    def agent_list_retrieval(self, agent_list, record_prefix="auto", service="auto"):
+        _ = (agent_list, record_prefix, service)
+
+    def bulk_record_retrieval(self, record_identifier, record_list):
+        _ = (record_identifier, record_list)
+
+    def build_key_list(self, agent_identifier, aik_list, service="auto"):
+        _ = (agent_identifier, aik_list, service)
+
+    def record_read(self, attestation_record_list, agent_identifier, service="auto"):
+        _ = (attestation_record_list, agent_identifier, service)
+
+    def _get_record_type(self, service):
+        if service != "auto":
+            self.svc = service
+
+        if self.svc == "registrar":
+            return "registration"
+
+        return "attestation"
+
+    def _get_certs_path(self, override_service=None):
+
+        if override_service:
+            self.svc = override_service
+
+        if not self.key_tls_pub:
+            (_, self.key_tls_priv, _, _), _ = web_util.get_tls_options(self.svc, logger=logger)
+
+            self.key_tls_pub = self.key_tls_priv.replace("-private", "-public")
+
+            if self.tl_url:
+                with open(self.key_tls_priv, "rb") as fp:
+                    self.priv_key = crypto.rsa_import_privkey(fp.read())
+
+    def _record_encode(self, record_object):
+        """Encodes record"""
+
+        _encoded_record_object = record_object
+
+        if self.rcd_fmt == "pickle":
+            _encoded_record_object = pickle.dumps(record_object)
+
+        if self.rcd_fmt == "json":
+            _encoded_record_object = json.dumps(record_object, indent=4)
+            if self.rcd_enc == "base64":
+                _encoded_record_object = str.encode(_encoded_record_object)
+
+        if self.rcd_enc == "base64":
+            _encoded_record_object = base64.b64encode(_encoded_record_object)
```

Comment:
So there is one serializer (`rcd_fmt`) and one encoder (`rcd_enc`). Checking the code is not clear if the serializer is optional, but the first sentence in like 77 suggest that it is optional (so the `_encoded_record_object` is always set).  In that case the base64 will fail, as it is not a valid serializer.

IMHO this code should change and force that the serializer is set.

Reply:
I think this still needs to be addressed.

Reply:
There is value on being able to inspect the contents of, say, `redis` persistent store using a browser and looking at a json object which can be read by an human while debugging.